test/uController.SourceGenerator.Tests/IntegrationTests.cs

```diff
@@ -812,28 +812,84 @@ public static object[][] AcceptsMetadataActions
     {
         get
         {
+            var implicitFromFormFile = 
+            $$"""
+            app.MapPost("/fileupload", (IFormFile file) =>
+            {
+                return $"Uploaded {file.Name}";
+            });
+            """;
+
+            var implicitFromFormCollection =
+            $$"""
+            app.MapPost("/formpost", (IFormCollection formCollection) =>
+            {
+                return $"Uploaded {formCollection.Count} files";
+            });
+            """;
+
+            var explicitFromFormFile = 
+            $$"""
+            app.MapPost("/fileupload", ([FromForm] IFormFile file) =>
+            {
+                return $"Uploaded {file.Name}";
+            });
+            """;
+
+            var explicitFromFormCollection =
+            $$"""
+            app.MapPost("/formpost", ([FromForm] IFormCollection formCollection) =>
+            {
+                return $"Uploaded {formCollection.Count} files";
+            });
+            """;
+
             return new[]
             {
-                new[] { ImplicitFromBodyActions[0][0], typeof(Todo), "application/json" },
-                new[] { ImplicitFromBodyActions[1][0], typeof(ITodo), "application/json" },
-                new[] { ImplicitFromBodyActions[2][0], typeof(TodoStruct), "application/json" },
-                new[] { ExplicitFromBodyActions[0][0], typeof(Todo), "application/json" }
+                new[] { ImplicitFromBodyActions[0][0], typeof(Todo), new[] { "application/json" }},
+                new[] { ImplicitFromBodyActions[1][0], typeof(ITodo), new[] { "application/json" }},
+                new[] { ImplicitFromBodyActions[2][0], typeof(TodoStruct), new[] { "application/json" }},
+                new[] { ExplicitFromBodyActions[0][0], typeof(Todo), new[] { "application/json" }},
+                new[] { (object)implicitFromFormFile, typeof(IFormFile), new[] { "multipart/form-data" }},
+                new[] { (object)implicitFromFormCollection, typeof(IFormCollection), new[] { "multipart/form-data" }},
+                new[] { (object)explicitFromFormFile, typeof(IFormFile), new[] { "multipart/form-data" }},
+                new[] { (object)explicitFromFormCollection, typeof(IFormCollection), new[] { "multipart/form-data" }}
             };
         }
     }
 
     [Theory]
     [MemberData(nameof(AcceptsMetadataActions))]
-    public async Task PopulatesAcceptsMetadataForRequestBody(string source, Type expectedType, string expectedContentType)
+    public async Task PopulatesAcceptsMetadataForRequestBody(string source, Type expectedType, string[] expectedContentTypes)
     {
-        var endpoint = await GetEndpoint(source);
+        var serviceProviderIsService = new ServiceProviderIsService();
```

Comment:
This is built in.

Reply:
Ah -- that's right. We're using the WAB for everything.